src/arrayMethodSort.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        let compareResult;
+
+        if (compareFunction) {
+          compareResult = compareFunction(this[i], this[j]) > 0;
+        } else {
+          compareResult = String(this[i]) > String(this[j]);
```

Comment:
You can create a separate variable for a default function and set it as a default value for compareFunction